apps/meteor/app/api/server/v1/moderation.ts

```diff
@@ -73,9 +73,12 @@ API.v1.addRoute(
 
 			const { count = 50, offset = 0 } = await getPaginationItems(this.queryParams);
 
-			const user = await Users.findOneById(userId, { projection: { _id: 1 } });
-			if (!user) {
-				return API.v1.failure('error-invalid-user');
+			const existingReport = await ModerationReports.findOne(
+				{ 'message.u._id': userId, '_hidden': { $ne: true } },
+				{ projection: { _id: 1 } },
+			);
```

Comment:
Move this to the model.

Reply:
Thank you! I will create a new method to handle this.